src/neoxp/Extensions/Extensions.cs

```diff
@@ -266,5 +267,48 @@ static bool TryGetNEP6Wallet(string path, string password, ProtocolSettings sett
                 }
             }
         }
+        
+        
+        private static JObject[] MakeFindStatesParams(
+            UInt256 rootHash,
+            UInt160 scriptHash,
+            ReadOnlySpan<byte> prefix,
+            ReadOnlySpan<byte> from = default (ReadOnlySpan<byte>),
+            int? count = null)
+        {
+            JObject[] jobjectArray = new JObject[count.HasValue ? 5 : 4];
+            jobjectArray[0] = (JObject) rootHash.ToString();
+            jobjectArray[1] = (JObject) scriptHash.ToString();
+            jobjectArray[2] = (JObject) Convert.ToBase64String(prefix);
+            jobjectArray[3] = (JObject) Convert.ToBase64String(from);
+            if (count.HasValue)
+                jobjectArray[4] = (JObject) (double) count.Value;
+            return jobjectArray;
+        }
+        public static async Task<ExpressRpcFoundStates> ExpressFindStatesAsync(this RpcClient rpcClient, UInt256 rootHash, UInt160 contractScriptHash, ReadOnlyMemory<byte> prefix, ReadOnlyMemory<byte> from = default, int? pageSize = null)
+        {
+            ExpressRpcFoundStates? states = null;
+            var start = from.ToArray();
+            while (true)
+            {
+                var @params = MakeFindStatesParams(rootHash, contractScriptHash, prefix.Span, start, pageSize);
+                var response = await rpcClient.RpcSendAsync("findstates", @params).ConfigureAwait(false);
+                var foundStates = ExpressRpcFoundStates.FromJson(response);
```

Comment:
Might make more sense to process the JSON here directly rather than deserializing into a class like ExpressRpcFoundStates. We only need `truncated` and `results` from `response.